spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/functions/BucketFunction.java

```diff
@@ -226,9 +231,11 @@ public String canonicalName() {
 
     @Override
     public Integer produceResult(InternalRow input) {
-      return input.isNullAt(NUM_BUCKETS_ORDINAL) || input.isNullAt(VALUE_ORDINAL)
-          ? null
-          : invoke(input.getInt(NUM_BUCKETS_ORDINAL), input.getUTF8String(VALUE_ORDINAL));
+      if (input.isNullAt(NUM_BUCKETS_ORDINAL) || input.isNullAt(VALUE_ORDINAL)) {
+        return null;
+      } else {
+        return invoke(input.getInt(NUM_BUCKETS_ORDINAL), input.getInt(VALUE_ORDINAL));
```

Comment:
Looks like a typo. Shall it still be `input.getUTF8String()`?

Reply:
Good catch. Updated!